samcli/commands/cloud/remote_invoke_context.py

```diff
@@ -105,20 +108,28 @@ def _populate_resource_summary(self) -> None:
         if not self._stack_name and not self._resource_id:
             raise InvalidRemoteInvokeParameters("Either --stack-name or --resource-id parameter should be provided")
 
-        if not self._resource_id:
-            # no resource id provided, list all resources from stack and try to find one
-            self._resource_summary = self._get_single_resource_from_stack()
-            self._resource_id = self._resource_summary.logical_resource_id
-            return
-
-        if not self._stack_name:
-            # no stack name provided, resource id should be physical id so that we can use it
-            self._resource_summary = self._get_from_physical_resource_id()
-            return
-
-        self._resource_summary = get_resource_summary(
-            self._boto_resource_provider, self._boto_client_provider, self._stack_name, self._resource_id
-        )
+        try:
+            if not self._resource_id:
+                # no resource id provided, list all resources from stack and try to find one
+                self._resource_summary = self._get_single_resource_from_stack()
+                self._resource_id = self._resource_summary.logical_resource_id
+                return
+
+            if not self._stack_name:
+                # no stack name provided, resource id should be physical id so that we can use it
+                self._resource_summary = self._get_from_physical_resource_id()
+                return
+
+            self._resource_summary = get_resource_summary(
+                self._boto_resource_provider, self._boto_client_provider, self._stack_name, self._resource_id
+            )
+        except ClientError as ex:
+            error_code = get_client_error_code(ex)
```

Comment:
do we care about any other errors?

Reply:
I think `ValidationError` should be enough here.